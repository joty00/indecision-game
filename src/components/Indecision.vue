<template>
  <img v-if="img" :src="img" alt="bg">
  <div class="bg-dark"></div>

  <div class="indecision-container">
    <input 
    v-model="question"
    type="text" 
    placeholder="Make me a question">
    <p>Remeber the interogation sign (?) at the end.</p>
    <div v-if="isValidQestion">
        <h2>{{ question }}</h2>
        <h1>{{ answer }}</h1>
    </div>
  </div> 
</template>

<script>
    export default {
        data() {
            return {
                question: null,
                answer: null,
                img: null,
                isValidQestion: false,
            }
        },
        methods: {
            async getAnswer() {
                this.answer = 'Thinking...'
                /*If your app is in spanish use this line:
                this.answer = 'Pensando...'*/   
                
                const {answer, image} = await fetch('https://yesno.wtf/api').then( r => r.json())
                /*If your app is in spanish use this line:
                this.answer = answer === 'yes' ? "Si!" : "No!"*/
                this.answer = answer
                this.img = image
            }
        },
        watch: {
            question( value, oldValue) {
                this.isValidQestion = false
                if(!value.includes('?')) return
                this.isValidQestion = true
                this.getAnswer()
            }
        }
    }
</script>

<style>
    img, .bg-dark {
        height: 100vh;
        left: 0px;
        max-height: 100%;
        max-width: 100%;
        position: fixed;
        top: 0px;
        width: 100vw;
    }
    .bg-dark {
        background-color: rgba(0, 0, 0, 0.4);
    }
    .indecision-container {
        position: relative;
        z-index: 99;
    }
    
    input {
        width: 250px;
        padding: 10px 15px;
        border-radius: 5px;
        border: none;
    }
    input:focus {
        outline: none;
    }
    p {
        color: white;
        font-size: 20px;
        margin-top: 0px;
    }
    h1, h2 {
        color: white;
        z-index: 999;
    }
    
    h2 {
        margin-top: 150px;
    }
</style>