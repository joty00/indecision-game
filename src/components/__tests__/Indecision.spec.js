import { describe, it, expect, beforeEach, test } from 'vitest'

import { mount, shallowMount } from '@vue/test-utils'
import Indecision from '../Indecision.vue'

describe('Indecision', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Indecision)
  })

  test('debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

})