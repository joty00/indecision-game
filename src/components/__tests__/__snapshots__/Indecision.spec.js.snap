// Vitest Snapshot v1

exports[`Indecision > debe de hacer match con el snapshot 1`] = `
"<!--v-if-->
<div class=\\"bg-dark\\"></div>
<div class=\\"indecision-container\\"><input type=\\"text\\" placeholder=\\"Make me a question\\">
  <p>Remeber the interogation sign (?) at the end.</p>
  <!--v-if-->
</div>"
`;
